core/payment-driver/glmsync/src/driver.rs

```diff
@@ -329,7 +342,28 @@ Mind that to be eligible you have to run your app at least once on testnet -
             .into_iter()
             .map(|allocation| allocation.remaining_amount)
             .sum();
-        Ok(msg.amount <= (account_balance - total_allocated_amount))
+
+        // NOTE: `wallet::get_tx_fee` accepts an _recipient_ address which is unknown at the moment
+        // so the _sender_ address is provider. This might bias fee calculation, because transaction
+        // to new account is little more expensive.
+        let tx_fee_cost = wallet::get_tx_fee(&msg.address, network).await?;
+        let total_txs_cost = BigDecimal::from(*ALLOCATE_FOR_N_TRANSACTIONS) * tx_fee_cost;
+        let max_allocation_surcharge =
+            BigDecimal::min(BigDecimal::from(*MAX_ALLOCATION_SURCHARGE), total_txs_cost);
+
+        log::info!(
+            "Allocation validation: \
+            - allocating: {:.5}\n \
```

Comment:
nice addition!

i think it will still print nicely without `\n` in there, maybe you can replace it with `, `

Reply:
Yep. Avoid newlines in log messages.